src/main/java/org/folio/service/mapping/MappingServiceImpl.java

```diff
@@ -39,18 +44,30 @@ public MappingServiceImpl() {
     }
     List<String> records = new ArrayList<>();
     ReferenceData referenceData = referenceDataProvider.get(jobExecutionId, connectionParams);
-    List<Rule> rules = ruleFactory.create(mappingProfile);
+    List<Rule> rules = getRules(mappingProfile, connectionParams);
     for (JsonObject instance : instances) {
       String record = runDefaultMapping(instance, referenceData, rules);
       records.add(record);
     }
     return records;
   }
 
-  private String runDefaultMapping(JsonObject instance, ReferenceData referenceData, List<Rule> rules) {
+  private String runDefaultMapping(JsonObject instance, ReferenceData referenceData, List<Rule> rules)                                                                                          {
     EntityReader entityReader = new JPathSyntaxEntityReader(instance);
     RecordWriter recordWriter = new MarcRecordWriter();
     return this.ruleProcessor.process(entityReader, recordWriter, referenceData, rules);
   }
 
+  private List<Rule> getRules(MappingProfile mappingProfile, OkapiConnectionParams params) {
+    List<Rule> rulesFromConfig = configurationsClient.getRulesFromConfiguration(params);
+    return CollectionUtils.isEmpty(rulesFromConfig) ? ruleFactory.create(mappingProfile) : appendRulesFromProfile(rulesFromConfig, mappingProfile);
+  }
+
+  private List<Rule> appendRulesFromProfile(List<Rule> rulesFromConfig, MappingProfile mappingProfile) {
+    if (mappingProfile != null && isNotEmpty(mappingProfile.getTransformations())) {
+      rulesFromConfig.addAll(ruleFactory.buildByTransformations(mappingProfile.getTransformations()));
```

Comment:
Can you please add a debug message here saying we are using overridden rules with transformations

Reply:
sure